server/src/main/java/org/candlepin/util/RdbmsExceptionTranslator.java

```diff
@@ -54,4 +56,31 @@ public boolean isUpdateHadNoEffectException(RollbackException sqlException) {
         return false;
     }
 
+    public boolean isConstraintViolationDuplicateEntry(PersistenceException pe) {
+        log.debug("Translating {}", pe);
+
+        if (pe.getCause() != null &&
+            pe.getCause() instanceof ConstraintViolationException) {
+
+            ConstraintViolationException cve = (ConstraintViolationException) pe.getCause();
+            log.info("ConstraintViolationException error code:" + cve.getErrorCode());
+
+            //MySQL error code ER_DUP_ENTRY
+            //http://dev.mysql.com/doc/refman/5.7/en/error-messages-server.html
+            if (cve.getSQLState() != null &&
+                cve.getSQLState().equals("23000") &&
+                cve.getErrorCode() == 1062) {
+                return true;
+            }
+
+            //Postgres error code DUPLICATE OBJECT
+            //https://www.postgresql.org/docs/8.3/static/errcodes-appendix.html
+            if (cve.getSQLState() != null &&
+                cve.getSQLState().equals("23505")) {
+                return true;
+            }
+        }
```

Comment:
I don't have an Oracle box to test this. And AFAIK we do not run on Oracle right now. Can I just leave a TODO here?